kong/clustering/services/sync/init.lua

```diff
@@ -53,10 +53,31 @@ function _M:init_worker()
     return
   end
 
-  -- sync to CP ASAP
-  assert(self.rpc:sync_once(FIRST_SYNC_DELAY))
+  local worker_events = assert(kong.worker_events)
 
-  assert(self.rpc:sync_every(EACH_SYNC_DELAY))
+  -- if rpc is ready we will start to sync
+  worker_events.register(function(capabilities_list)
+    local has_sync_v2
+
+    -- check cp's capabilities
+    for _, v in ipairs(capabilities_list) do
+      if v == "kong.sync.v2" then
+        has_sync_v2 = true
+        break
+      end
+    end
+
+    -- cp does not support kong.sync.v2
+    if not has_sync_v2 then
+      return
+    end
```

Comment:
I think here we need to report a serious error like `crit` or `alert` level, showing that KONG DP is trying to use sync.v2 to synchronize configs, but CP does not support.

Because if we configure kong.rpc/kong.sync, we will not run sync.v1 synchronization, so this DP may never get configuration from CP

Reply:
If cp disabled sync v2, sync v1 will start, see clustering/data_plane.lua's changes.

Reply:
we need to add a `warning` to report that we will use full sync v1 if CP does not support sync.v2. See v1 logic, it will check if CP does support v2, if CP does not support v2, it will run v1 full sync.

<img width="514" alt="image" src="https://github.com/user-attachments/assets/281a5367-c394-4843-bc97-f2811dce3a74" />